consensus/polybft/bridge_event_relayer.go

```diff
@@ -0,0 +1,460 @@
+
+// sendTx is a goroutine that listens to the event channel and sends the appropriate transactions
+func (ber *bridgeEventRelayerImpl) sendTx() {
+	for {
+		select {
+		case <-ber.quitCh:
+			return
+		case event := <-ber.eventCh:
+			switch event := event.(type) {
+			case *contractsapi.SignedBridgeMessageBatch:
+				if err := ber.sendBridgeMessageBatch(event); err != nil {
+					ber.logger.Error("error occurred while sending bridge message batch transaction", "error", err)
+				}
+			case *contractsapi.SignedValidatorSet:
+				if err := ber.sendCommitValidatorSet(event); err != nil {
+					ber.logger.Error("error occurred while sending commit validator set transaction", "error", err)
+				}
+			default:
+				ber.logger.Error("unknown event type", "event", event)
+			}
+		}
+	}
+}
+
+// sendBridgeMessageBatch sends bridge message batch execute transaction to the external and internal chains
+func (ber *bridgeEventRelayerImpl) sendBridgeMessageBatch(event *contractsapi.SignedBridgeMessageBatch) error {
+	var (
+		txRelayer          = ber.internalTxRelayer
+		destinationChainID = event.Batch.DestinationChainID.Uint64()
+		to                 = ber.bridgeConfig[destinationChainID].InternalGatewayAddr
+		exists             bool
+	)
+
+	if event.Batch.DestinationChainID.Cmp(ber.internalChainID) != 0 {
+		txRelayer, exists = ber.externalTxRelayers[destinationChainID]
+		if !exists {
+			return fmt.Errorf("tx relayer for chain %d not found", destinationChainID)
+		}
+
+		to = ber.bridgeConfig[destinationChainID].ExternalGatewayAddr
+	}
+
+	input, err := (&contractsapi.ReceiveBatchGatewayFn{
+		Batch:     event.Batch,
+		Signature: event.Signature,
+		Bitmap:    event.Bitmap,
+	}).EncodeAbi()
+	if err != nil {
+		return err
+	}
+
+	txn := types.NewTx(types.NewLegacyTx(
+		types.WithFrom(ber.key.Address()),
+		types.WithTo(&to),
+		types.WithInput(input),
+	))
+
+	receipt, err := txRelayer.SendTransaction(txn, ber.key)
+	if err != nil {
+		return fmt.Errorf("failed to execute batch transaction on chain: %d, and gateway contract: %s. Error: %w",
+			destinationChainID, to, err)
+	}
+
+	ber.logger.Info("sent batch transaction",
+		"destinationChainID", destinationChainID,
+		"gatewayAddr", to,
+		"status", types.ReceiptStatus(receipt.Status),
+		"txHash", receipt.TransactionHash,
+		"blockNumber", receipt.BlockNumber,
+	)
+
+	return nil
+}
+
+// sendCommitValidatorSet sends commit validator set transaction to the external and internal chains
+func (ber *bridgeEventRelayerImpl) sendCommitValidatorSet(event *contractsapi.SignedValidatorSet) error {
+	input, err := (&contractsapi.CommitValidatorSetBridgeStorageFn{
+		NewValidatorSet: event.NewValidatorSet,
+		Signature:       event.Signature,
+		Bitmap:          event.Bitmap,
+	}).EncodeAbi()
+	if err != nil {
+		return err
+	}
+
+	for chainID, txRelayer := range ber.externalTxRelayers {
+		// send commit validator set transaction to the external chains
+		to := ber.bridgeConfig[chainID].ExternalGatewayAddr
```

Comment:
Extract to a variable and reuse it in the rest of function:

```suggestion
                bridgeCfg := ber.bridgeConfig[chainID]
		to := bridgeCfg.ExternalGatewayAddr
```